src/fondant/pipeline/pipeline.py

```diff
@@ -185,25 +184,59 @@ def __init__(
         self.resources = resources or Resources()
 
     @classmethod
-    def from_component_yaml(cls, path, **kwargs) -> "ComponentOp":
+    def from_component_yaml(cls, path, fields=None, **kwargs) -> "ComponentOp":
         if cls._is_custom_component(path):
             component_dir = Path(path)
         else:
             component_dir = cls._get_registry_path(str(path))
+
         component_spec = ComponentSpec.from_file(
             component_dir / cls.COMPONENT_SPEC_NAME,
         )
 
+        # If consumes is not defined in the pipeline, we will try to infer it
+        if kwargs.get("consumes", None) is None:
+            kwargs["consumes"] = cls._infer_consumes(component_spec, fields)
+
         image = Image(
             base_image=component_spec.image,
         )
+
         return cls(
             image=image,
             component_spec=component_spec,
             component_dir=component_dir,
             **kwargs,
         )
 
+    @classmethod
+    def _infer_consumes(cls, component_spec, dataset_fields):
+        """Infer the consumes section of the component spec."""
+        if component_spec.consumes_is_defined is False:
+            msg = (
+                "The consumes section of the component spec is not defined. "
+                "Cannot infer consumes."
+            )
+            logger.info(msg)
+            return None
+
+        # Component has consumes and additionalProperties, we will load all dataset columns
```

Comment:
Wondering about the case here where there's both a schema and `additionalProperties`. But I guess this is fine for now. We might want to have more logic here once we add support for `additionalProperties` schemas instead of just the boolean.